front/lib/auth.ts

```diff
@@ -800,12 +800,24 @@ export class Authenticator {
   hasPermission(acls: ACLType[], permission: Permission): boolean {
     // For each acl, does the user belongs to a group that has the permission?
     return acls.every((acl) =>
-      this.groups().some((group) =>
-        groupHasPermission(acl, permission, group.id)
+      hasResourcePermission(
+        acl,
+        this.getNonNullableWorkspace(),
+        permission,
+        this.groups(),
+        this.role()
       )
     );
   }
 
+  canAdministrate(acls: ACLType[]): boolean {
+    return this.hasPermission(acls, "admin");
+  }
+
+  canList(acls: ACLType[]): boolean {
```

Comment:
Ideally we kill canList with :fire: in favor of just canRead or canAdministrate

Reply:
Yep for now used to abstract `canRead() || canAdministrate()` but will be killed in a follow up PR.